Task 21:Completed

onxas opened the pull request:
Created Service and Controller to get and change info about User

onxas committed:
TASK-21: User Endpoit

Added User endpoit
Started user preferences endpoint

onxas committed:
TASK-4: Cnage User Info

Updated UserPage Cotroller
Fixed bugs

onxas committed:
TASK-21:Updated UserPage endpoint

Updated User Page Contrroller
Created service to user page

onxas committed:
TASK-21: Completed

Created UserPAge Service
Updated UserPage Controller

onxas committed:
TASK-23: Completed

Created UserPAge Service
Updated UserPage Controller

BrineFast commented on be/src/main/java/com/savethepet/controller/HomeController.java:
```diff
@@ -34,33 +34,28 @@
+            newUser.setName(userFromOauth.getAttribute("real_name"));
+            newUser.setYandexId(principal.getName());
             userRepo.save(newUser);
         }
```
In the `User` class we have `@AllArgsConstructor`. So maybe it's better to use this to do everything in one line rather than 4

BrineFast commented on be/src/main/java/com/savethepet/controller/UserPageController.java:
```diff
@@ -0,0 +1,116 @@
+                                .email(user.getEmail())
+                                .img(user.getImg())
+                                .build());
+    }
```
This is unnecessary, Spring will independently send `200`, if everything is normal. So just return the DTO object

BrineFast commented on be/src/main/java/com/savethepet/controller/UserPageController.java:
```diff
@@ -0,0 +1,116 @@
+        userRepo.save(changedUser);
+        HttpHeaders responseHeaders = new HttpHeaders();
+        responseHeaders.setLocation(URI.create(rerouteURL + "/user/" + id.toString()));
+        return new ResponseEntity(responseHeaders, HttpStatus.OK);
```
Business logic and why are we redirecting somewhere?

BrineFast commented on be/src/main/java/com/savethepet/service/UserPageService.java:
```diff
@@ -0,0 +1,45 @@
+                throw new RuntimeException("Unknown client registration id= " + registrationId);
+        }
+    }
+}
```
These methods may be obvious, but nobody canceled `javadoc`

BrineFast commented on be/src/main/java/com/savethepet/service/UserPageService.java:
```diff
@@ -0,0 +1,45 @@
+                return userRepo.findByFacebookId(username).orElseThrow(() ->
+                        new UserNotFoundException("User with Oauth2 id=" + username + "not found"));
+            default:
+                throw new RuntimeException("Unknown client registration id= " + registrationId);
```
Don't you think that 4 identical strings, except for 1 variable, don't look good if you constantly rewrite them?

onxas committed:
TASK-21: Created service

save commit to work in outher pc

onxas committed:
TASK-21: Created service

save commit to work in outher pc

onxas committed:
Revert "TASK-21: Created service"

This reverts commit 5b2161c01b300e01941fd06f23e507b128822773.

onxas committed:
Revert "TASK-21: Created service"

This reverts commit 5b2161c01b300e01941fd06f23e507b128822773.

onxas committed:
Task-21: updating

save to another pc

onxas committed:
Task-21: updating

save to another pc

onxas committed:
TASK-21: Added Oauth2 Changes

Added endpoints to
Link new OAuth2
Delete OAuth2
Update service layer to UserPage

onxas committed:
TASK-21: Added Oauth2 Changes

Added endpoints to
Link new OAuth2
Delete OAuth2
Update service layer to UserPage

onxas committed:
TASK-21:Updated Service layer

onxas committed:
TASK-24:Updated Service layer

onxas committed:
Merge branch 'TASK-21' of https://github.com/BrineFast/SaveThePet into TASK-21

onxas committed:
Merge branch 'master' into TASK-21

BrineFast commented on be/src/main/java/com/savethepet/controller/OAuth2Controller.java:
```diff
@@ -0,0 +1,139 @@
+    )
+    @Transactional
+    @GetMapping("/be/oauth/registration")
+    public ResponseEntity<Void> addUserFromOauth2(@ApiIgnore Principal principal) {
```
Please throw exceptions

BrineFast commented on be/src/main/java/com/savethepet/controller/UserPageController.java:
```diff
@@ -0,0 +1,138 @@
+    public UserInfoDTO getUserInfo(@PathVariable("user_id") Long id) {
+        User user = userPageService.getUserById(id);
+        HttpHeaders responseHeaders = new HttpHeaders();
+        responseHeaders.setLocation(URI.create(rerouteURL + "/user/" + id.toString()));
```
Useless

BrineFast commented on be/src/main/java/com/savethepet/controller/UserPageController.java:
```diff
@@ -0,0 +1,138 @@
+    private String rerouteURL;
+
+    @Autowired
+    private UserRepo userRepo;
```
Useless

BrineFast commented on be/src/main/java/com/savethepet/service/UserPageService.java:
```diff
@@ -0,0 +1,122 @@
+        }
+        userRepo.save(user);
+    }
+}
```
Comments

BrineFast commented on be/src/main/java/com/savethepet/service/UserPageService.java:
```diff
@@ -0,0 +1,122 @@
+        userRepo.save(changedUser);
+    }
+
+    public User getUserByOauth2ClientRegistrationId(String registrationId, String username) {
```
Bad variable name `username` for auth token

onxas committed:
TASK-21:Corrections

onxas committed:
TASK-21:Corrections & Bug Fixes

onxas committed:
Update UserPageService.java

onxas committed:
TASK-21:Documentation and features

Added admin account and config to it
Fixed user table in db
Added Start App and Auth docs

onxas committed:
TASK-21: Updated Doc

Updated How Auth
